### ISSUE TITLE
Undefined symbols for architecture arm64

### ISSUE THREAD

**Issue by ASHFAQPATWARI** [arnesson/cordova-plugin-firebase#949]
**Describe the bug**

**To Reproduce**

**Expected behavior**

**Screenshots**

**Console Logs**
If applicable, provide console logs showing the errors being reported.

**Plugin Version**
Run `cordova plugin list` to find out what version of cordova-plugin-firebase you are running with

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

**Additional context**

**Comment by yeahstyle** [comment]
Same issue here:

```
Undefined symbols for architecture armv7:
  "_FIRAuthStateDidChangeInternalNotificationUIDKey", referenced from:
      -[FIRAuth possiblyPostAuthStateChangeNotification] in FirebaseAuth(FIRAuth.o)
  "_OBJC_CLASS_$_FIRComponent", referenced from:
      objc-class-ref in FirebaseAuth(FIRAuth.o)
  "_OBJC_CLASS_$_FIRComponentType", referenced from:
      objc-class-ref in FirebaseAuth(FIRAuth.o)
  "_OBJC_CLASS_$_FIRComponentContainer", referenced from:
      objc-class-ref in FirebaseAuth(FIRAuth.o)
ld: symbol(s) not found for architecture armv7
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

cordova 8.1.2 (cordova-lib@8.1.1)
cordova ios 4.5.0
xcode Version 10.1 (10B61)

**Comment by yeahstyle** [comment]
> I have been having the same issue, what version of the plugin was working for you previously? Does it still work if the version is reverted?

I've tried 2.0.0 and I have the same issue

**Comment by yeahstyle** [comment]
alright- in my case this plugin was conflicting with another plugin:
<plugin name="phonegap-plugin-push" spec="^2.1.3" />

removed that one and this one worked just fine. It looks like I don't need that one anymore anyways since this one handles push notifications as well

**Comment by dwolner** [comment]
what other plugin were you using that conflicted?

**Comment by yeahstyle** [comment]
> what other plugin were you using that conflicted?

oh! forgot to mention it- duh!

https://github.com/phonegap/phonegap-plugin-push

**Comment by misterjojo** [comment]
How did you manage to find out which plugin conflicted with this one ? I'm facing the same issue but i don't use phonegap-plugin-push

**Comment by ASHFAQPATWARI** [comment]
@dwolner I am on 2.0.5. I have tried reverting back to many previous versions but none of them work now.

@misterjojo I am also not using phonegap-plugin-push and i am still facing the issue.

**Comment by vuhrmeister** [comment]
Same here … not using phonegap-plugin-push

**Comment by yeahstyle** [comment]
> How did you manage to find out which plugin conflicted with this one ? I'm facing the same issue but i don't use phonegap-plugin-push

Since both plugins do the same (they both require firebase) I just though it'd make sense to remove the other plugin.

**Comment by vuhrmeister** [comment]
Found out that cordova-plugin-googleplus causes the conflict. After removing it build on iOS works.

**Comment by dwolner** [comment]
Interesting, I was able to fix this by reverting the plugin version back to v1.1.4 which I had used on a previous project. I am also using cordova-plugin-googleplus and cordova-plugin-code-push but did not try experimenting with removing them.

**Comment by vuhrmeister** [comment]
I just went back to version 1.05 … this is working again. But rather not the version we want to use :(

**Comment by dwolner** [comment]
Yeah agreed, now running into some API issues requiring a setTimeout that I am not happy about and are most likely fixed in 2.0:
```window.FirebasePlugin.grantPermission(() => {
                        setTimeout(() => {
                            window.FirebasePlugin.subscribe('general')
                        }, 2000)
                    })```
Need to pinpoint the build issue in 2.0 but I am not sure where to start, maybe with the major changes.

**Comment by Yuripetusko** [comment]
Same issue here and I have `cordova-plugin-googleplus` also

**Comment by tmyers97** [comment]
Same issue with cordova-plugin-googleplus as well. Has anyone had any luck besides downgrading?

**Comment by WeihangTang** [comment]
> > what other plugin were you using that conflicted?
> 
> oh! forgot to mention it- duh!
> 
> https://github.com/phonegap/phonegap-plugin-push

thanks, life saver !!!

**Comment by lAndresul** [comment]
In my case it was the Google Plus add-on, I just needed to update and the problem was solved. I think maybe the problem is related to this: https://developers.google.com/android/guides/releases#may_23_2018, so if you have more than one add-on that uses Google Play services, you should update all or maybe if it's your case for your convenience you need to make a downgrade

**Comment by hiepxanh** [comment]
I hate this bug, no solution

**Comment by Sroose** [comment]
> ugin, I just needed to update and the problem was solved. I think maybe the problem is related to this: https://developers.google.com/android/guides/releases#may_23_2018, so if you have more than one plugin that uses Google Play services, you should update all your plugins, or maybe if it's your case for convenience

Hi Andres, to which versions did you upgrade exactly?

**Comment by Sroose** [comment]
FYI, [SO question for workaround](https://stackoverflow.com/questions/56132435/solve-incompatibility-cordova-firebase-and-google-login-in-ios) created

**Comment by devotebest** [comment]
I got same problem when use "cordova-plugin-firebase" and "cordova-plugin-googleplus".
what is solution?

**Comment by manojalwisnz** [comment]
Just remove phonegap-plugin-push and run again
